homeassistant/components/knx/storage/entity_store_schema.py

```diff
@@ -106,6 +113,18 @@ def optional_ga_schema(key: str, ga_selector: GASelector) -> VolDictType:
     }
 )
 
+SENSOR_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_ENTITY): BASE_ENTITY_SCHEMA,
+        vol.Required(DOMAIN): {
+            vol.Required(CONF_GA_SENSOR): GASelector(state_required=True),
+            vol.Required(CONF_VALUE_TYPE): str,
```

Comment:
```suggestion
            vol.Required(CONF_GA_SENSOR): GASelector(
                write=False,
                state_required=True,
                dpt=[],
            ),
```
The interesting thing for UI configurable sensor entities will be how to pass and validate available `dpt` from `xknx.dpt` up to the UI and back.
`value_type` can be omitted then.

Reply:
Got your point. That means DPT is a mandatory field when it comes to a sensor, right?

Reply:
The [group address selector](https://github.com/XKNX/knx-frontend/blob/bcfe4cea5686b977158f3568b4bef25fcccabcbb/src/components/knx-group-address-selector.ts#L223) does return it this way. It is not yet tuned to support such many DPTs though.

I was also thinking of serializing all selectors (including valid dpts for each key - which are many for sensor 🙃) to pass it to the frontend (like https://github.com/home-assistant-libs/voluptuous-serialize) so the schema is only hardcoded in the backend, and doesn't have to be replicated in the frontend - it didn't happen yet though.

Reply:
I believe it makes the most sense to handle mapping, exposing, and validation of DPTs directly within the xknx library. I've created a PR XKNX/xknx#1629 to address this, aiming to centralize the KNX-specific logic in one repository. What are your thoughts?

Reply:
left a comment there as well. HA SensorEntity has different constraints than xknx Sensor.
I'd like to keep HA specifics in the HA knx integration and move the few things we still have in xknx out of there in the long run. Eg. `ha_device_class` should move into the integration, as well as unit of measurement so we can use HA enums.